packages/renderer/src/lib/ui/Input.svelte

```diff
@@ -29,45 +30,58 @@ async function onClear() {
 }
 </script>
 
-<div
-  class="flex flex-row items-center px-1 py-1 group bg-charcoal-500 border-[1px] border-charcoal-500 {$$props.class ||
-    ''}"
-  class:hover:bg-charcoal-900="{!readonly && !disabled}"
-  class:focus-within:bg-charcoal-900="{!readonly && !disabled}"
-  class:hover:rounded-md="{!readonly && !disabled}"
-  class:focus-within:rounded-md="{!readonly && !disabled}"
-  class:border-b-purple-500="{!readonly && !disabled}"
-  class:hover:border-purple-400="{!readonly && !disabled}"
-  class:focus-within:border-purple-500="{!readonly && !disabled}"
-  class:border-b-charcoal-100="{readonly || disabled}">
-  <slot name="left" />
-  <input
-    bind:this="{element}"
-    on:input
-    class="grow px-1 outline-0 bg-charcoal-500 text-sm placeholder:text-gray-700 overflow-hidden"
-    class:text-white="{!disabled}"
-    class:text-gray-700="{disabled}"
-    class:group-hover:bg-charcoal-900="{!readonly && !disabled}"
-    class:group-focus-within:bg-charcoal-900="{!readonly && !disabled}"
-    class:group-hover-placeholder:text-gray-900="{!readonly && !disabled}"
-    name="{name}"
-    type="text"
-    disabled="{disabled}"
-    readonly="{readonly}"
-    required="{required}"
-    placeholder="{placeholder}"
-    id="{id}"
-    aria-label="{$$props['aria-label']}"
-    aria-invalid="{$$props['aria-invalid']}"
-    bind:value="{value}" />
-  {#if clearable}
-    <button
-      class="px-1 cursor-pointer text-gray-700 group-hover:text-gray-900 group-focus-within:text-gray-900"
-      class:hidden="{!value || readonly || disabled}"
-      aria-label="clear"
-      on:click="{onClear}">
-      <Fa icon="{faXmark}" />
-    </button>
+<div class="flex flex-col w-full">
+  <div
+    class="flex flex-row w-full items-center px-1 py-1 group bg-charcoal-500 border-[1px] border-charcoal-500 {$$props.class ||
+      ''}"
+    class:hover:bg-charcoal-900="{!readonly && !disabled}"
```

Comment:
It looks like we should find another pattern for specifying classes as I think we reached a state where it's unreadable/unmaintainable
due to all these conditions

Reply:
Agreed. I was letting it go a bit b/c I didn't know if we still had other requirements, but I think we've got most of the inputs and it's time to simplify.

Any ideas other than extracting a couple variables for the more complicated/merged states? Ok with separate PR?

Reply:
Can be follow-up of course, required changes was for the test